components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -114,35 +114,29 @@ public boolean isTableExists() throws CertificateManagementException {
         String getCertsQuery = SQLConstants.CertificateConstants.GET_CERTIFICATES;
 
         try {
-            try {
-                connection = APIMgtDBUtil.getConnection();
-                initialAutoCommit = connection.getAutoCommit();
-                connection.setAutoCommit(false);
-                connection.commit();
-                preparedStatement = connection.prepareStatement(getCertsQuery);
-                preparedStatement.setInt(1, tenantId);
-                resultSet = preparedStatement.executeQuery();
+            connection = APIMgtDBUtil.getConnection();
+            initialAutoCommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+            connection.commit();
+            preparedStatement = connection.prepareStatement(getCertsQuery);
+            preparedStatement.setInt(1, tenantId);
+            resultSet = preparedStatement.executeQuery();
 
-                while (resultSet.next()) {
-                    certificateMetadataDTO = new CertificateMetadataDTO();
-                    certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
-                    certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
-                    certificates.add(certificateMetadataDTO);
-                }
-            } catch (SQLException e) {
-                if (log.isDebugEnabled()) {
-                    log.debug("Error occurred while retrieving certificate metadata. ", e);
-                }
-                handleException("Error while retrieving certificates.", e);
-            } finally {
-                connection.setAutoCommit(initialAutoCommit);
-                APIMgtDBUtil.closeStatement(preparedStatement);
-                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+            while (resultSet.next()) {
+                certificateMetadataDTO = new CertificateMetadataDTO();
+                certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
+                certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
+                certificates.add(certificateMetadataDTO);
             }
         } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error occurred in finally block while retrieving certificate metadata. ", e);
+                log.debug("Error occurred while retrieving certificate metadata. ", e);
             }
+            handleException("Error while retrieving certificates.", e);
+        } finally {
+            APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
+            APIMgtDBUtil.closeStatement(preparedStatement);
+            APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
```

Comment:
Do we need to explicitly call APIMgtDBUtil.closeStatement? APIMgtDBUtil.closeAllConnections does that as well.